packages/components/pie-toast-provider/src/defs.ts

```diff
@@ -1,3 +1,54 @@
-// TODO - please remove the eslint disable comment below when you add props to this interface
-// eslint-disable-next-line @typescript-eslint/no-empty-interface
-export interface ToastProviderProps {}
+import { type ToastProps } from '@justeattakeaway/pie-toast';
+
+import { type ComponentDefaultProps } from '@justeattakeaway/pie-webc-core';
+
+export const PRIORITY_ORDER: { [x: string]: number } = {
+    'error-actionable': 1,
+    error: 2,
+    'warning-actionable': 3,
+    'success-actionable': 4,
+    'info-actionable': 5,
+    'neutral-actionable': 6,
+    warning: 7,
+    success: 8,
+    info: 9,
+    neutral: 10,
+};
+
+export interface ExtendedToastProps extends ToastProps {
+    /**
+     * Triggered when the user interacts with the close icon or when the toast auto dismiss.
+     */
+    onPieToastClose?: () => void;
```

Comment:
Is there a reason we decided to use function props here rather than emitting events? Also what happens if there is more than one toast?

Reply:
yes, users will interface with the component using the [toaster singleton](https://github.com/justeattakeaway/pie/blob/main/apps/pie-storybook/stories/pie-toast-provider.stories.ts#L89) which accepts an object for all toast props.

we are only passing props to the toast component which does emit the event internally so no new events were created.

Reply:
this is an exmaple in aperture if it helps 
https://github.com/justeattakeaway/pie-aperture/pull/228/files#diff-b30a824bb84fa94b7a399a009b0bc5d1e67556bddec47fa26dc717e2de07a0faR21-R24 

options is the global configuration of all toasts controlled by the provider but individual calls of the toaster supports that too.